components/post_view/post/index.js

```diff
@@ -15,7 +15,7 @@ function mapStateToProps(state, ownProps) {
     const detailedPost = ownProps.post || {};
 
     return {
-        post: getPost(state, detailedPost.id),
```

Comment:
This'll probably be fine given how rare this case is, but we shouldn't generally do this when PureComponents are involved since, if getPost is returning undefined, the post prop is being set to a new empty object on every store change, causing this Post component to re-render.

Reply:
Actually, that's a really good point. Let me update it to handle in the component itself

Reply:
Done